lua/resession/init.lua

```diff
@@ -69,11 +81,47 @@ M.get_current = function()
   return tab_sessions[tabpage] or current_session
 end
 
----Detach from the current session
-M.detach = function()
-  current_session = nil
-  local tabpage = vim.api.nvim_get_current_tabpage()
-  tab_sessions[tabpage] = nil
+---Detach from a session
+---@param name? string
+---@param opts? resession.DetachOpts
+---@param target_tabpage? integer
+M.detach = function(name, opts, target_tabpage)
+  opts = vim.tbl_extend("keep", opts or {}, {
+    notify = true,
+  })
+  if target_tabpage then
+    name = tab_sessions[target_tabpage]
+    if not name then
+      error(string.format("No session attached to tab %s", target_tabpage))
```

Comment:
These errors feel a bit too harsh. I would expect that there are some situations where we would want to "detach from session named <x>, or no-op". If some feedback is required, a common practice is to return a boolean that indicates if the detach operation was performed.